mimic/rest/identity_api.py

```diff
@@ -250,6 +250,43 @@ def get_users_details(self, request):
             "updated": seconds_to_timestamp(time.time())
         }))
 
+    @app.route('/v2.0/users/<string:user_id>/OS-KSADM/credentials',
+               methods=['GET'])
+    def get_user_credentials_osksadm(self, request, user_id):
+        """
+        Support, such as it is, for the credentials call.
+
+        reference: http://developer.openstack.org/api-ref-identity-v2-ext.html
+            #listCredentials
+        """
+        if user_id in self.core.sessions._userid_to_session:
+            username = self.core.sessions._userid_to_session[user_id].username
+            password = '9d5ed678fe57bcca610140957afab571'  # echo -n B | md5sum
+            apikey = '0d61f8370cad1d412f80b84d143e1257'  # echo -n C | md5sum
+            return json.dumps(
+                {
+                    'credentials': [
+                        {
+                            'passwordCredentials': {
```

Comment:
Reading my comment back I guess it doesn't make a lot of sense.

The ability to give back a plain-text password means that the system must _store_ plain-text passwords.  A well-configured system should only ever be storing password hashes, making it impossible to give back a plain-text password.  That's what I'm hoping we're doing, and the fact that I can't find a way to get `passwordCredentials` back on my own accounts strongly suggests that we are.

Given the way that API keys work in OpenStack, it does actually have to store _them_ in plain text (which, IMHO, is still a design error: API keys should be generated, given back to the user, then hashes of them should be stored, and if you need a new API key you should have to request a new one, not look up the old one, but it's a less serious error because presumably randomly-generated credentials can't be shared with _other_ systems).